src/export-grants.ts

```diff
@@ -37,44 +42,93 @@ function parseNumber(n: number, decimals: number) {
-        p.released = parseNumber(p.released, decimals)
+async function fillEnactingTxData(proposal: Grant & ProposalParsed) {
+  try {
+    const url = `https://api.covalenthq.com/v1/${NetworkID[Network.ETHEREUM]}/transaction_v2/${proposal.enacting_tx}/?key=${COVALENT_API_KEY}`
+    const json = await fetchURL(url)
+    if (json.error) {
+      console.log(`Error trying to get contract data for proposal ${proposal.id}`, json)
+      return
+    }
+    const data: APITransactions = json.data
+    const transactions: TransactionItem = data.items[0]
+
+    transactions.log_events.forEach(logEvent => {
+      const decodedLogEvent = logEvent.decoded
+      if (decodedLogEvent) {
+        if (decodedLogEvent.name === 'Transfer') {
+          const txMatchesBeneficiary = decodedLogEvent.params.some(param =>
+              param.name === ParamName.To && String(param.value).toLowerCase() === proposal.grant_beneficiary.toLowerCase());
+          if (txMatchesBeneficiary) {
+            proposal.token = DECIMALS[logEvent.sender_address][0]
+            const decimals: number = DECIMALS[logEvent.sender_address][1]
+            proposal.tx_date = logEvent.block_signed_at
+            findTxAmount(decodedLogEvent, proposal, decimals);
+          }
+        }
+      }
+    })
+  } catch (e) {
+    console.log(`Error trying to get contract data for proposal ${proposal.id}`, e)
+  }
+}
 
-        p.releasable = await contract.methods.releasableAmount().call()
-        p.releasable = parseNumber(p.releasable, decimals)
+async function main() {
+  // Get Governance dApp Proposals
+  const proposals: GrantProposal[] = PROPOSALS.filter(p => p.type === GovernanceProposalType.GRANT)
 
-        const contractStart:number = await contract.methods.start().call()
-        const contractDuration:number = await contract.methods.duration().call()
-        const contractEndsTimestamp = Number(contractStart) + Number(contractDuration)
-        p.vesting_finish_at = new Date(contractEndsTimestamp * 1000)
+  for (const proposal of proposals) {
+    proposal.grant_category = proposal.configuration.category
+    proposal.grant_tier = proposal.configuration.tier.split(':')[0]
+    proposal.grant_size = proposal.configuration.size
+    proposal.grant_beneficiary = proposal.configuration.beneficiary
 
-      } catch (e){
-        console.log(`Error trying to get vesting data for proposal ${p.id}`, e)
-      }
+    if (proposal.vesting_address) {
+      await fillVestingContractData(proposal);
+    }
+    if (proposal.enacting_tx) {
+      await fillEnactingTxData(proposal);
+    }
+    if(proposal.vesting_address === null && proposal.enacting_tx === null){
+      console.log(`A proposal without vesting address and enacting tx has been found. Id ${proposal.id}`)
     }
   }
 
   console.log(proposals.length, 'grants found.')
 
   saveToJSON('grants.json', proposals)
-  saveToCSV('grants.csv', proposals, [
+  await saveToCSV('grants.csv', proposals, [
```

Comment:
Should we await this?

Reply:
Yes, it's an async method so we want to know that it finishes properly before returning